# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/webkitgtk41/hotfix-cmake.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- webkitgtk-2.46.5/Source/cmake/WebKitMacros.cmake.vanilla	2026-09-21 19:22:21.514718840 +0300
+++ webkitgtk-2.46.5/Source/cmake/WebKitMacros.cmake	2026-09-21 19:30:54.313039664 +0300
@@ -277,6 +277,6 @@
         get_property(_linked_into GLOBAL PROPERTY ${framework}_LINKED_INTO)
 
         # See if the target is linking a framework that the specified framework is already linked into
+	if (NOT ${_linked_into})
-        if ((NOT _linked_into) OR (${framework} STREQUAL ${_linked_into}) OR (NOT ${_linked_into} IN_LIST ${_target}_FRAMEWORKS))
             list(APPEND ${_target}_PRIVATE_LIBRARIES WebKit::${framework})
 
@@ -286,7 +285,15 @@
             if (TARGET ${framework} AND ${framework}_LIBRARY_TYPE STREQUAL "OBJECT")
                 list(APPEND ${_target}_PRIVATE_LIBRARIES $<TARGET_OBJECTS:${framework}>)
             endif ()
-        endif ()
+       endif ()
+       if (${_linked_into})
+               if ((${framework} STREQUAL ${_linked_into}) OR (NOT ${_linked_into} IN_LIST ${_target}_FRAMEWORKS))
+                       list(APPEND ${_target}_PRIVATE_LIBRARIES WebKit::${framework})
+                       if (TARGET ${framework} AND ${framework}_LIBRARY_TYPE STREQUAL "OBJECT")
+                               list(APPEND ${_target}_PRIVATE_LIBRARIES $<TARGET_OBJECTS:${framework}>)
+                       endif ()
+               endif ()
+       endif ()
     endforeach ()
 endmacro()

